test/e2e/operator/garden/common.go

```diff
@@ -35,20 +35,38 @@ const namespace = "garden"
 var (
 	parentCtx     context.Context
 	runtimeClient client.Client
+
+	extensionProviderLocal *operatorv1alpha1.Extension
 )
 
 var _ = BeforeSuite(func() {
 	logf.SetLogger(logger.MustNewZapLogger(logger.InfoLevel, logger.FormatJSON, zap.WriteTo(GinkgoWriter)))
-})
-
-var _ = BeforeEach(func() {
-	parentCtx = context.Background()
 
 	restConfig, err := kubernetes.RESTConfigFromClientConnectionConfiguration(&componentbaseconfig.ClientConnectionConfiguration{Kubeconfig: os.Getenv("KUBECONFIG")}, nil, kubernetes.AuthTokenFile)
 	Expect(err).NotTo(HaveOccurred())
 
 	runtimeClient, err = client.New(restConfig, client.Options{Scheme: operatorclient.RuntimeScheme})
 	Expect(err).NotTo(HaveOccurred())
+
+	// TODO(timuthy): Remove this special handling as soon as extensions provider a proper deletion procedure. Planned for release v1.103 or v1.104.
```

Comment:
Can you update the comment and specify "proper deletion procedure"? It's not clear to me what this means.

Reply:
The extension resources, especially the managed resource for the admission controller should be deleted if the `garden` is deleted.

I have extended the comment, but maybe @timuthy wants to take a look himself.